src/app/modules/main/communities/minting/models/token_item.nim

```diff
@@ -0,0 +1,92 @@
+import strformat
+
+type
+  TokenType* {.pure.} = enum
+    #ERC20,
+    ERC721
+
+type
+  MintingState* {.pure.} = enum
+    InProgress,
+    Minted
+
+type
+  TokenItem* = ref object
+    tokenType: TokenType
+    tokenAddress: string
+    name: string
+    description: string
+    icon: string
```

Comment:
By `icon` you mean the `artwork` itself? If yes, I would suggest to use another more explicity naming for it, like `artwork` or `image`, maybe?!